arch-install.bash

```diff
@@ -1,9 +1,31 @@
 #!/bin/bash
-
 set -euo pipefail
 
 shopt -s extglob globstar nullglob
 
+while getopts "g:i:" opt; do
+  case $opt in
+  g) git_repo=$OPTARG ;;
+  *) usage ;;
+  esac
+done
+
+if [[ -v git_repo ]]; then
+  export git_repo
+  shift $((OPTIND - 1))
+  encoded_command=""
+  for arg in "$@"; do
+    encoded_command+=";$(base64 <<<"$arg")"
+  done
+  encoded_command=${encoded_command:1}
+  export encoded_command
+fi
```

Comment:
There should be an `elif` branch here testing whether `"$#" == 0` since operands aren't allowed if `-g` hasn't been specified.

Reply:
The `shift $((OPTIND - 1))` should be moved out of the if body, otherwise `$#` won't have the correct value.